Redesign: New icons

barklund opened the pull request:
## Context

See #6299 for todo.

### QA

<!--
Not all changes require manual QA.
-->

<!-- ignore-task-list-start -->
- [x] This is a non-user-facing change and requires no QA
<!-- ignore-task-list-end -->

## Checklist

<!-- Check these after PR creation -->

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

<!--
Please reference the issue(s) this PR addresses.
No URLs, just the issue numbers.
Use "Fixes #123" if it fixes an issue.
Example:

Fixes #123
Partially addresses #456
See #789
-->

Fixes #6299

Commit:
Moved dashboard specific icons out of design system

BrittanyIRL reviewed (dismissed):
Thanks for doing this, makes sense to me - especially after talking this morning about focusing on making the design system the shared parts of the project.

Commit:
Replaced all icons with new ones

This has many consequences as the new icons are differently boxed compared to the old ones. I will try to fix in subsequent commits.

Commit:
Fixed icon styling

Commit:
Added small icons, fixed dropdown chevron

Commit:
Fixed cross icon in snackbar

Commit:
Changed all icon colors to dynamic and added storybook demo

Commit:
Fixed inspector icon usage and storybook

Commit:
Fixed plus icon in color picker

Commit:
Fixed xs chevron icons after alignment

Commit:
Fixed plus icons in color preset empty panel

Commit:
Fixed icons in preset panel header

Commit:
Swapped two icons

Commit:
Fixed design in help center including new icons

Commit:
Reversed help center chevron direction

BrittanyIRL commented on assets/src/design-system/icons/stories/index.js:
```diff
@@ -80,173 +71,26 @@ export const AllIcons = () => {
 
-export const Action = () => {
+export const ColorfulIcons = () => {
+  const colors = ['blue', 'hotpink', 'rebeccapurple', 'lightgreen', 'red'];
```
💅

BrittanyIRL commented on assets/src/edit-story/components/helpCenter/toggle/index.js:
```diff
@@ -127,16 +118,19 @@ function Toggle({
-        <ChevronIcon />
-      </Icon>
+      <Label>{__('Help Center', 'web-stories')}</Label>
+      <Space />
```
This feels odd? Could this not be added to the label as padding?

barklund replied:
I like this approach - it also makes RTL simpler, as the divs just go in the other order instead of having to switch margin or padding from right to left. We've used it in a few places too. But I'm okay with changing it.

BrittanyIRL replied:
ah, if RTL is the reason then I'm ok with it. I was unaware that this was a standard in the editor.

Commit:
Renamed icon

Commit:
Merge branch 'main' into task/6299-ds-icons

Commit:
Updated instructions

BrittanyIRL reviewed (approved):
Thanks for updating all of these at once. 

To me there's a little bit of duplicate here with direction differences, which if that's what we want to be the standard I am open to but I want to mention it. It would be simple enough to have `Arrow` and eliminate ArrowDown, ArrowRight, ArrowUp - same thing for ChevronUp vs Down, since that will probably almost always change based on a prop it wouldn't be that big of a deal to rely on css for the direction update. WDYT? 

Design system, dashboard, help and prepublish updates all checkout. I found 4 things that needed to be updated, made a PR for the updates to this branch to save you the busy work: https://github.com/google/web-stories-wp/pull/6319 if you agree with them. 

Not worried about the dropdown chevron because I have an update for that with search anyway since those styles changed. 

Approving with comments since we're in different timezones and I'd rather not slow this down.

BrittanyIRL committed:
Made some icon adjustments

Make checkmark bigger in design system (used on dashboard/settings and telemetry banner), make toggle checkmark bigger (only exists in storybook), tweak prepublish alert icon that shows up on publisher or update button to be better size, move space for help center toggle to be padding instead of a div. (#6319)

Commit:
Edited icon instructions

Commit:
Merge branch 'task/6299-ds-icons' of github.com:google/web-stories-wp into task/6299-ds-icons

barklund commented on docs/design-system.md:
```diff
@@ -67,9 +67,17 @@ Icons are separated from components and already exported as `ReactComponent` . Y
+Icons are all located directly in the `design-system/icons/` folder. A few things to note:
 
-1. Export your SVG & compress it (example: [https://jakearchibald.github.io/svgomg/](https://jakearchibald.github.io/svgomg/))
-2. Add the file to the appropriate directory in [design-system/icons](../assets/src/design-system/icons/index.js)
-3. Export it from the index of that directory as a ReactComponent
+- All icons are 32x32px and should never be resized or scaled in any way.
+- Icons are named after their symbol, not their function. So a ⚙️  icon is named `Gear`, and not `Settings` or some other function, it might serve.
+- If an icon has several sizes, denote this as a suffix as `GearLarge`, `GearTiny` etc.
+- If an icon has a border, name it either just as `GearOutline` or as the shape of the border - e.g. `GearTriangle` if it's a gear inside a triangular border.
+- Try to group similar icons alphabetically, so name icons e.g. `MagnifierPlus` and `MagnifierMinus` rather than `PlusMagnifier` and `MinusMagnifier`.
```
@BrittanyIRL, can you approve of this change?

BrittanyIRL replied:
Thanks for adding documentation! Like the note about trying to maintain alphabetization, think this makes having the icons flat in one directory more manageable.

Commit:
Fixed some icon colors

samwhale commented on docs/design-system.md:
```diff
@@ -67,9 +67,17 @@ Icons are separated from components and already exported as `ReactComponent` . Y
+
+The process for adding an icon is as follows:
+
+1. Export your SVG from Figma by selecting the 32x32px icon boundary box and selecting _"Export"_ in the right panel
+2. Add the file to directory [design-system/icons](../assets/src/design-system/icons/index.js)
+3. Export it from the index of that directory as a ReactComponent (please alphabetize the list, thanks!)
```
Should we add a step here to compress the svg first?

BrittanyIRL replied:
optimization is happening by default in webpack - https://github.com/google/web-stories-wp/pull/6287#discussion_r573089953

barklund replied:
That supposed to happen automagically in the build pipeline, but I'm actually not sure if we still do this.

@swissspidy, can you confirm that this happens? I only see `svgr` in the [Webpack config](https://github.com/google/web-stories-wp/blob/main/webpack.config.cjs#L95), but no mention of `svgo`.

[This doc](https://react-svgr.com/docs/options/#svgo) indicates that `svgo` is standard when using `svgr`, so I assume it works?

swissspidy replied:
It should still be the default, yes. There's also a `.svgo.yml` config file for it that - at least when I last tested it - was applied to SVGs.

You could quickly test this yourself by removing those configs and re-building the app. It should cause the icons in the app to appear rather broken :-).

If it doesn't, then there's a bug.

samwhale commented on assets/src/edit-story/components/helpCenter/toggle/index.js:
```diff
@@ -127,16 +112,18 @@ function Toggle({
+        ) : (
+          <Icon isOpen={isOpen}>
+            <ChevronIcon />
+          </Icon>
```
So now the chevron only shows after the notifications are gone. Is that intended?
![image](https://user-images.githubusercontent.com/22185279/107675525-66f0a900-6c55-11eb-8110-2f52b150e7ce.png)

samwhale replied:
![image](https://user-images.githubusercontent.com/22185279/107675678-8c7db280-6c55-11eb-928a-5aaa9a7ad94e.png)

BrittanyIRL replied:
it is, designs were updated very recently! https://xwp.slack.com/archives/C01ESBB8ZN1/p1612994877142300

barklund replied:
Yes, the designs have been updated to [this](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=5663%3A3546). Isn't it lovely with a moving target? 😁 

But I know see, that it has actually updated again since this implementation for the narrow viewport. Here "Help Center" is replaced by just the question mark icon followed by either the unread count or the chevron.

BrittanyIRL replied:
oh heck, i thought it was that on narrow viewport it's just 'help' + the chevron or badge depending on if there's unread tips? Either way, this update's not going in until after the release is cut so if it's off a little bit when this is merged it's ok because we can just have a folllow up ticket/pr to actually address that. I feel like adjusting this FTUE here is just scope creep.

barklund replied:
This will actually go in, as this mainly updates icons in the design system, and only very few icons in the actual editor - and none of those are changed visually, they're merely updated in sizing and bounding box.

barklund replied:
Implemented the new designs in 629038057. Feel free to take them for a spin - I think it looks pretty good. Also, see follow-up discussion [here](https://xwp.slack.com/archives/C01ESBB8ZN1/p1613098933195200).

samwhale reviewed (approved):
Icons in storybook look good. Help center looks good - but wondering if the chevron should still be shown when there are unread notifications

samwhale commented on assets/src/design-system/components/toggle/index.js:
```diff
@@ -35,9 +35,9 @@ const CIRCLE_FINAL_POSITION =
-const ICON_LEFT_POSITION = 30.5;
+const ICON_WIDTH = 18;
+const ICON_TOP_POSITION = 2;
+const ICON_LEFT_POSITION = 26;
```
💭 do you like the checkmark in the current position or up 1 pixel?

![checkmark-position](https://user-images.githubusercontent.com/22185279/107675797-adde9e80-6c55-11eb-9e97-fb5b0faa4e4b.gif)

samwhale replied:
To me it feels like it's better centered when it's slightly higher (maybe even 0.5 px), but I'll leave that to your judgement

barklund replied:
Great catch! I've forwarded the question to UX.

samitron7 replied:
Can we move it one pixel up. looks more centered.

barklund replied:
Fixed in e08c625c6.

Commit:
Fixed icon size in storybook

Commit:
Merge branch 'main' into task/6299-ds-icons

Commit:
Renamed some icons

Commit:
Fixed position and css for ds toggle checkmark

Commit:
Scaled page layouts icon back up

Commit:
Updated styling of + in color picker text

barklund commented on assets/src/edit-story/components/panels/design/preset/colorPreset/colorPresetActions.js:
```diff
@@ -214,10 +220,10 @@ function ColorPresetActions({ color, pushUpdate }) {
               mapping={{
-                i: <Add />,
+                b: <Strong />,
               }}
             >
-              {__('Click <i></i> to save a color', 'web-stories')}
+              {__('Click <b>+</b> to save a color', 'web-stories')}
```
@miina, while looking to fix this, I noticed that the actual **+** used in the design is a real text plus just in a bigger font and in white. So I did the same here and it looks pretty similar:

| Design | Implementation |
|-|-|
| ![Screen Shot 2021-02-11 at 20 49 07](https://user-images.githubusercontent.com/637548/107721439-a2699280-6caa-11eb-9f69-444c5fb132ea.png) | ![Screen Shot 2021-02-11 at 20 48 57](https://user-images.githubusercontent.com/637548/107721441-a3022900-6caa-11eb-83a1-a2a45667d4d3.png) |

miina replied:
Sounds good to me, the approach. Looks like it needs a bit more room though and looks thicker than on the design. Maybe reduce the font size to 20 and/or make it inline-block to assign fixed width? (I can look at it within #6208 as well though, not critical!)

barklund replied:
Let's deploy as-is for 1.4 and we can fix in #6208 when redoing the component anyway.

Commit:
Updated help center toggle button layout

barklund commented on assets/src/edit-story/components/library/panes/pageLayouts/pageLayoutsIcon.js:
```diff
@@ -23,10 +23,10 @@ import { __ } from '@web-stories-wp/i18n';
-const StyledLayoutIcon = styled(Layout)({
-  transform: 'scale(0.71)',
+const StyledLayoutIcon = styled(Icons.Box4Alternate)({
+  transform: 'scale(1.41)',
```
This is a purely temporary (and quite magic - `sqrt(2)` even!) value until #6207 is implemented. This will go out for 1.4, but should be fine.

Commit:
Exit debug
